contracts/vaults/VoltzVault.sol

```diff
@@ -278,167 +124,201 @@ contract VoltzVault is IVoltzVault, IntegrationVault {
 
-            IPeriphery.SwapPeripheryParams memory params = IPeriphery.SwapPeripheryParams({
-                marginEngine: _marginEngine,
-                isFT: (opts.notionalForTrade < 0),
-                notional: (opts.notionalForTrade < 0) ? (-opts.notionalForTrade).toUint256() : opts.notionalForTrade.toUint256(),
-                sqrtPriceLimitX96: sqrtPriceLimitX96,
-                tickLower: _currentPosition.low,
-                tickUpper: _currentPosition.high,
-                marginDelta: 0
+        _currentPosition = ticks;
+    }
+
+    /// @notice Divide a given time in seconds by the number of seconds in a year
+    /// @param timeInSecondsAsWad A time in seconds in Wad (i.e. scaled up by 10^18)
+    /// @return timeInYearsWad An annualised factor of timeInSeconds, also in Wad
+    function _accrualFact(uint256 timeInSecondsAsWad)
+        internal
+        pure
+        returns (uint256 timeInYearsWad)
+    {
+        timeInYearsWad = timeInSecondsAsWad.div(SECONDS_IN_YEAR_IN_WAD);
+    }
+    
+    function _unwindAndExitCurrentPosition() internal {
+        Position.Info memory _currentPositionInfo = _marginEngine.getPosition(
+            address(this),
+            _currentPosition.tickLower,
+            _currentPosition.tickUpper
+        );
+
+        if (_currentPositionInfo.variableTokenBalance != 0) {
+            bool _isFT = _currentPositionInfo.variableTokenBalance < 0;
+
+            IVAMM.SwapParams memory _params = IVAMM.SwapParams({
+                recipient: address(this),
+                amountSpecified: _currentPositionInfo.variableTokenBalance,
+                sqrtPriceLimitX96: _isFT
+                    ? TickMath.MIN_SQRT_RATIO + 1
+                    : TickMath.MAX_SQRT_RATIO - 1,
+                tickLower: _currentPosition.tickLower,
+                tickUpper: _currentPosition.tickUpper
             });
 
-            periphery.swap(params);
+            IVAMM _vamm = _marginEngine.vamm();
+
+            _vamm.swap(_params);
+        } 
+
+        _currentPositionInfo = _marginEngine.getPosition(
+            address(this),
+            _currentPosition.tickLower,
+            _currentPosition.tickUpper
+        );
+
+        uint256 positionMarginRequirementInitial = _marginEngine.getPositionMarginRequirement(
+            address(this),
+            _currentPosition.tickLower,
+            _currentPosition.tickUpper,
+            false
+        );
+
+        int256 marginToWithdraw;
+        if (_currentPositionInfo.variableTokenBalance != 0) {
+            // keep k * initial margin requirement, withdraw the rest
+            // need to track to redeem the rest at maturity
+            marginToWithdraw = _currentPositionInfo.margin - (_k * positionMarginRequirementInitial).toInt256();
+            trackedPositions.push(_currentPosition);
+        } else if (_currentPositionInfo.fixedTokenBalance > 0) {
+            // withdraw all margin
```

Comment:
@avniculae is this the redundant scenario right?

Reply:
Indeed it is